archive/xar_header.ksy

```diff
@@ -0,0 +1,58 @@
+meta:
+  id: xar_header
+  title: eXtensible ARchiver
+  file-extension:
+    - pkg
+    - xar
+    - xip
+  xref:
+    mime: application/x-xar
+    pronom: fmt/600
+    wikidata: Q1093556
+  license: CC0-1.0
+  ks-version: 0.9
+  encoding: UTF-8
+  endian: be
+doc: |
+  From Wikipedia:
+
+  "XAR (short for eXtensible ARchive format) is an open source file archiver
+  and the archiver’s file format. It was created within the OpenDarwin project
+  and is used in macOS X 10.5 and up for software installation routines, as
+  well as browser extensions in Safari 5.0 and up."
+
+  It should be noted that there is a different version from Apple uses the
+  same version number as the (unmaintained) version described here, but that
+  supports additional checksums, but doesn't have checksum_name.
+versions
+doc-ref: https://github.com/mackyle/xar/wiki/xarformat
+seq:
+  - id: magic
+    contents: 'xar!'
+  - id: header_size
```

Comment:
I don't see any `header` specified that occupues exactly the count of bytes that is stored in this field.

Reply:
It is in the header itself.

Reply:
It can be moved into an own subtype, which size is determined by this field within it.

Reply:
This would probably solve my issue of the variants as well: if the header size is larger than 28, then I can switch to the format described here, otherwise to the format that Apple uses. The header of the Apple variant will (according to the source code), always set it to 28.